sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -490,6 +495,101 @@ def test_spline_transformer_n_features_out(
+    is passed and missing values are present in X."""
+    X_nan = np.array([[1, 1], [2, 2], [3, 3], [np.nan, 5], [4, 4]])
+    sample_weight = [1, 1, 1, 1, 1]
+    spline = SplineTransformer(knots="quantile", handle_missing="zeros")
+
+    msg = "Passing `sample_weight` to SplineTransformer when there are"
+    with pytest.raises(NotImplementedError, match=msg):
+        spline.fit(X_nan, sample_weight=sample_weight)
+
+
+@pytest.mark.skipif(
+    sp_version < parse_version("1.8.0"),
+    reason="The option `sparse_output` is available as of scipy 1.8.0",
+)
+@pytest.mark.parametrize(
+    "extrapolation", ["error", "constant", "linear", "continue", "periodic"]
+)
+@pytest.mark.parametrize("sparse_output", [False, True])
+def test_spline_transformer_handles_missing_values(extrapolation, sparse_output):
+    """Test that SplineTransformer handles missing values correctly.
+    We only test for knots="uniform", since for "quantile" the metrics are calculated
+    differently with nans present and a different result is thus expected.
+    """
+    X_nan = np.array([[1, 1], [2, 2], [3, 3], [np.nan, 5], [4, 4]])
+    X_nan_full_column = np.array([[np.nan, np.nan], [np.nan, 1]])
+    X = np.array([[1, 1], [2, 2], [3, 3], [4, 5], [4, 4]])
+
+    # check correct error message for handle_missing="error"
+    msg = "X contains missing values (np.nan) and SplineTransformer is configured with"
+    with pytest.raises(ValueError, match=re.escape(msg)):
+        spline = SplineTransformer(
+            degree=2,
+            n_knots=3,
+            handle_missing="error",
+            extrapolation=extrapolation,
+        )
+        spline.fit_transform(X_nan)
+
+    # check correct results for handle_missing="zeros"
+    spline = SplineTransformer(
+        degree=2,
+        n_knots=3,
+        handle_missing="zeros",
+        extrapolation=extrapolation,
+        sparse_output=sparse_output,
+    )
+
+    # check for generic invariants
+    X_nan_transform = spline.fit_transform(X_nan)
+    if sparse.issparse(X_nan_transform):
+        X_nan_transform = X_nan_transform.toarray()
+    assert (X_nan_transform >= 0).all()
+    assert (X_nan_transform <= 1).all()
+
+    # check `fit_transform` does the same as `fit` and `then` transform
+    X_nan_transform = spline.fit_transform(X_nan)
+    X_nan_fit_then_transformed = spline.fit(X_nan).transform(X_nan)
+    assert_allclose_dense_sparse(X_nan_transform, X_nan_fit_then_transformed)
+
+    # check that transform works as expected when the passed data has not the same
+    # shape as the training set array:
+    X_transformed_same_shape = spline.fit_transform(X_nan)[::2]
+    X_transformed_different_shapes = spline.fit(X_nan).transform(X_nan[::2])
+    assert_allclose_dense_sparse(
+        X_transformed_same_shape, X_transformed_different_shapes
+    )
+
+    # check that if X has a feature of all nans SplineTransformer works as usual
```

Comment:
```suggestion
    # Check that if X has a feature of all nans SplineTransformer encodes them all
    # as zeros as it would do for non-nan constant columns.
```